tipg/factory.py

```diff
@@ -1251,17 +1251,72 @@ async def tilejson(
 
             minzoom = minzoom if minzoom is not None else tms.minzoom
             maxzoom = maxzoom if maxzoom is not None else tms.maxzoom
+
+            bounds = [-180, -90, 180, 90]
+            if collection.extent and collection.extent.spatial:
+                bounds = collection.extent.spatial.bbox[0]
+
             tj = {
                 "minzoom": minzoom,
                 "maxzoom": maxzoom,
                 "name": collection.id,
+                "bounds": bounds,
                 "tiles": [tile_endpoint],
             }
             if bounds := collection.bounds:
                 tj["bounds"] = bounds
 
             return ORJSONResponse(tj)
 
+        @self.router.get(
+            "/collections/{collectionId}/{tileMatrixSetId}/viewer",
+            response_class=HTMLResponse,
+        )
+        @self.router.get(
+            "/collections/{collectionId}/viewer", response_class=HTMLResponse
+        )
+        def viewer_endpoint(
+            request: Request,
+            collection=Depends(self.collection_dependency),
+            tileMatrixSetId: Literal["WebMercatorQuad"] = Query(  # noqa
+                "WebMercatorQuad",
+                description="Identifier selecting one of the TileMatrixSetId supported (default: 'WebMercatorQuad')",
+            ),
```

Comment:
I choose to use the timvt viewer (using Maplibre) which only works with WebMercatorQuad TMS. The main issue is if a user will implement endpoints without WebMercatorQuad in the `supported_tms` 🤷

if we wanted to use leaflet + support multiple TMS we could use https://gist.github.com/vincentsarago/2fe69adda0b5089e9e6ca6bfe99bc7c9

Reply:
I think maplibre is just fine. This viewer is mostly for quick testing/verifying.